src/plugins/gstreamer/camerabin/camerabinimageprocessing.cpp

```diff
@@ -309,14 +313,20 @@ bool CameraBinImageProcessing::isParameterValueSupported(QCameraImageProcessingC
             g_object_get(G_OBJECT(m_session->cameraSource()), "supported-color-tones", &supported_effects, NULL);
 
             if (supported_effects) {
-                int effect_count = g_variant_n_children(supported_effects);
-
-                for (int i = 0; i < effect_count; i++) {
-                    GVariant *mode = g_variant_get_child_value(supported_effects, i);
-                    supportedEffects << m_filterMap.keys().at(m_filterMap.values().indexOf((GstPhotographyColorToneMode)g_variant_get_int32(mode)));
+                QMap<QCameraImageProcessing::ColorFilter, GstPhotographyColorToneMode>::const_iterator it = m_filterMap.find(filter);
+                if (it != m_filterMap.end()) {
+                    int effect_count = g_variant_n_children(supported_effects);
```

Comment:
I'm not 100% sure, GVariants have always been a bit of a mystery to me, but if this the property is an array of ints then `g_variant_get_fixed_array()` may be significantly more efficient. If it works))

Reply:
g_variant_get_fixed_array does work and based on documentation the current way of using g_variant_get_child_value causes some overhead.